labcontrol/db/process.py

```diff
@@ -3783,3 +3790,407 @@ def _generate_amplicon_prep_information(self):
+                    FROM labman.sequencing_process sp
+                    LEFT JOIN labman.sequencing_process_lanes spl USING (
+                        sequencing_process_id)
+                    LEFT JOIN labman.pool_composition_components pcc1 ON (
+                        spl.pool_composition_id =
+                        pcc1.output_pool_composition_id)
+                   LEFT JOIN labman.library_prep_shotgun_composition libprepcp2
+                        ON (
+                        pcc1.input_composition_id = libprepcp2.composition_id)
+                    LEFT JOIN labman.composition libprepcpcp2 ON (
+                        libprepcp2.composition_id =
+                        libprepcpcp2.composition_id)
+                    LEFT JOIN labman.library_prep_shotgun_process libpreppr2
+                        ON (libprepcpcp2.upstream_process_id =
+                        libpreppr2.process_id)
+                    LEFT JOIN labman.well libprepwell2 ON (
+                        libprepcpcp2.container_id = libprepwell2.container_id)
+                    LEFT JOIN labman.plate libprepplate2 ON (
+                        libprepwell2.plate_id = libprepplate2.plate_id)
+                    WHERE sequencing_process_id = %s)
+                """
+
+        with sql_connection.TRN as TRN:
+            TRN.add(sql, [self.id])
+
+            results = [dict(r) for r in TRN.execute_fetchindex()]
+
+            for d in results:
+                # assert that study_id and orig_name are only None when both
+                # are None; making this a control entry and not a sample entry
+                # (based on empirical data). However, check for both.
+                if d['study_id'] is None and d['orig_name'] is None:
+                    # adding this field allows the user to alter 'study_id'
+                    # and/or 'orig_name' downstream, w/out losing important
+                    # metadata. Also clearly names this property.
+                    d['is_control'] = True
+                else:
+                    d['is_control'] = False
+
+                # convert time from datetime objects to strings
+                # no timezone information appears with %z; Assume timestamp
+                # w/out timezone type in Pg.
+                # therefore, assume timestamps are local to San Diego and are
+                # PST/PDT as appropriate.
+                d['primer_date_i5'] =\
+                    d['primer_date_i5'].strftime(Process.get_date_format())
+                d['primer_date_i7'] =\
+                    d['primer_date_i7'].strftime(Process.get_date_format())
+
+                # instrument_model remains the same across all rows in this
+                # query.
+                d['instrument_model'] = inst_mdl
+
+                # note that the correct term is 'Kapa', not 'kappa'.
+                id = d['kappa_hyper_plus_kit_id']
+                d['kapa_hyper_plus_kit_lot'] = reagent[id]['external_lot_id']
```

Comment:
Both `kappa` (line 3966) and `kapa` (line 3967) are used. Would it be horrendous to standardize on one right now (even if incorrect for the time being)?

Reply:
Unfortunately, it's referenced as 'kappa' in its database column, so if we truly wanted to go one way right now, we'd probably be better off staying as 'kappa'. That would also make the column name in the prep file 'kappa' also, unless we want to break standardization for the very final product. That would be my vote.

Reply:
Ah okay

Reply:
Perhaps update #506 with a mention of this?